src/VolumeRenderSettings.ts

```diff
@@ -0,0 +1,243 @@
+import { Euler, Vector2, Vector3 } from "three";
+import Volume from "./Volume";
+import { Bounds } from "./types";
+
+/**
+ * Marks groups of related settings that may have changed.
+ */
+export enum SettingsFlags {
+  /** parameters: translation, rotation, scale, currentScale, flipAxes */
+  TRANSFORM = 1,
+  /** parameters: gammaMin, gammaLevel, gammaMax, brightness*/
+  CAMERA = 2,
+  /** parameters: showBoundingBox, boundingBoxColor */
+  BOUNDING_BOX = 4,
+  /** parameters: bounds, zSlice */
+  ROI = 8,
+  /** parameters: maskChannelIndex, maskAlpha */
+  MASK = 16,
+  /** parameters: density, specular, emissive, glossiness */
+  MATERIAL = 32,
+  /** parameters: resolution, useInterpolation, pixelSamplingRate, primaryRayStepSize, secondaryRayStepSize*/
+  SAMPLING = 64,
+  /** parameters: isOrtho, orthoScale, viewAxis, visible, maxProjectMode */
+  VIEW = 128,
+  ALL = Number.MAX_SAFE_INTEGER,
```

Comment:
the value of all should be 0xFFFFFFFF (the union of all single bits turned on).  I'm not sure what MAX_SAFE_INTEGER is.  Sometimes it can be clearer to write the flag constants as 0x1, 0x2, 0x4, 0x8, 0xF, 0x10, 0x20, 0x40, etc.

Reply:
even more explicit is `0b` notation where you can put the actual bit in position:
`var num8 = 0b001000; // binary`

Reply:
To be clear, the only important thing here is to know that ALL will actually have all bits turned on :)